src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,261 @@
+static_assert(PAL_GSS_C_MUTUAL_FLAG == GSS_C_MUTUAL_FLAG, "");
+static_assert(PAL_GSS_C_REPLAY_FLAG == GSS_C_REPLAY_FLAG, "");
+static_assert(PAL_GSS_C_SEQUENCE_FLAG == GSS_C_SEQUENCE_FLAG, "");
+static_assert(PAL_GSS_C_CONF_FLAG == GSS_C_CONF_FLAG, "");
+static_assert(PAL_GSS_C_INTEG_FLAG == GSS_C_INTEG_FLAG, "");
+static_assert(PAL_GSS_C_ANON_FLAG == GSS_C_ANON_FLAG, "");
+static_assert(PAL_GSS_C_PROT_READY_FLAG == GSS_C_PROT_READY_FLAG, "");
+static_assert(PAL_GSS_C_TRANS_FLAG == GSS_C_TRANS_FLAG, "");
+static_assert(PAL_GSS_C_DCE_STYLE == GSS_C_DCE_STYLE, "");
+static_assert(PAL_GSS_C_IDENTIFY_FLAG == GSS_C_IDENTIFY_FLAG, "");
+static_assert(PAL_GSS_C_EXTENDED_ERROR_FLAG == GSS_C_EXTENDED_ERROR_FLAG, "");
+static_assert(PAL_GSS_C_DELEG_POLICY_FLAG == GSS_C_DELEG_POLICY_FLAG, "");
+
+static_assert(PAL_GSS_COMPLETE == GSS_S_COMPLETE, "");
+static_assert(PAL_GSS_CONTINUE_NEEDED == GSS_S_CONTINUE_NEEDED, "");
+
+#if !(HAVE_GSS_SPNEGO_MECHANISM)
+static char gss_mech_value[] = "\x2b\x06\x01\x05\x05\x02"; // Binary representation of SPNEGO Oid (RFC 4178)
+#endif
+
+static void NetSecurity_HandleError(uint32_t majorStatus, GssBuffer** outBufferHandle, uint32_t* outBufferLength)
+{
+    assert(outBufferHandle != nullptr);
+    assert(outBufferLength != nullptr);
+    if (GSS_ERROR(majorStatus) || (*outBufferHandle)->length > UnsignedCast(std::numeric_limits<uint32_t>::max()))
+    {
+        uint32_t relBufferStatus;
+        gss_release_buffer(&relBufferStatus, *outBufferHandle);
+        *outBufferHandle = nullptr;
+        *outBufferLength = 0;
+    }
+    else
+    {
+        *outBufferLength = static_cast<uint32_t>((*outBufferHandle)->length);
+    }
+}
+
+extern "C" uint32_t NetSecurity_AcquireCredSpNego(uint32_t* minorStatus,
+                                                  GssName* desiredName,
+                                                  int32_t isInitiate,
+                                                  GssCredId** outputCredHandle)
+{
+    assert(minorStatus != nullptr);
+    assert(outputCredHandle != nullptr);
+    assert(isInitiate == 0 || isInitiate == 1);
+#if HAVE_GSS_SPNEGO_MECHANISM
+    gss_OID_set_desc gss_mech_spnego_OID_set_desc = {.count = 1, .elements = GSS_SPNEGO_MECHANISM};
+#else
+    gss_OID_desc gss_mech_spnego_OID_desc = {.length = 6, .elements = static_cast<void*>(gss_mech_value)};
+    gss_OID_set_desc gss_mech_spnego_OID_set_desc = {.count = 1, .elements = &gss_mech_spnego_OID_desc};
+#endif
+    gss_cred_usage_t credUsage = isInitiate ? GSS_C_INITIATE : GSS_C_ACCEPT;
+    return gss_acquire_cred(
+        minorStatus, desiredName, 0, &gss_mech_spnego_OID_set_desc, credUsage, outputCredHandle, nullptr, nullptr);
+}
+
+extern "C" uint32_t NetSecurity_DeleteSecContext(uint32_t* minorStatus, GssCtxId** contextHandle)
+{
+    assert(contextHandle != nullptr);
+    assert(minorStatus != nullptr);
+    return gss_delete_sec_context(minorStatus, contextHandle, GSS_C_NO_BUFFER);
```

Comment:
> Does the underlying function change the actual pointer?

Yes. It frees the resources pointed to by the actual pointer.
 The actual API exposed by the library is as follows:
  
              gss_delete_sec_context(OM_uint32 *minorStatus, gss_ctx_id_t *handle, gss_buffer_t outputHandle);

The typical usage is:

                   gss_ctx_id_t handle; // create an object on the stack.
                   gss_init_sec_contex( &status, &handle, other args);
                     // use context
                    gss_delete_sec_context(&status, &handle, &buffer);      

Since the ```gss_ctx_id_t``` type here is actually a pointer to a structure, we are seeing a double pointer.